Overhaul Fewest Moves (#678)
thewca/wca-regulations: wca-guidelines.md

- E2b+) [CLARIFICATION] A competitor may choose to stop an attempt early by handing in a solution before the time limit.
- E2c+) [CLARIFICATION] Although the judge may provide a standard sheet for competitors to submit solutions, a competitor may submit a solution on a different piece of paper. (Note that the piece of paper must come from the judge, according to [Regulation E3a](regulations:regulation:E3a).)
- E2c1++) [EXPLANATION] The competitor is not required to submit a paper with the competition name, round, and/or attempt number, but may wish to do so in order to help the organization team keep track of submitted solutions.
- E2c1+++) [ADDITION] If a judge finds a solution without a information to identify the attempt after the competitor has finished submitting it (e.g. while grading the attempt), they should find out the competitor who wrote the solution in order to assign them a DNF (rather than DNS). Note that the competitor must not be allowed to write information to identify the attempt at this point, so they must receive a DNF result.

> - E2c2+) [EXAMPLE] Examples that are not considered a single unambiguous move sequence: multiple solution sequences without exactly one clearly indicated solution, a solution with ambiguous symbols (e.g. a letter that looks similar to "B" and "R" but is not clearly one or the other), a solution with stars or long arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation), a move sequence that is not written out roughly as a series of lines in reading order, any arrangement of symbols that cannot be interpreted as an unambiguous move sequence. Exceptions to the examples above: small typographic corrections (e.g. blacking out a move and writing a move above that is clearly meant to replace it, moves written in subscript or superscript to fit between other moves).

Reviewer: Do we want the `moves written in subscript or superscript to fit between other moves`? This feels a bit weird to me, as it could turn out into `a move sequence that is not written out roughly as a series of lines in reading order` or `any arrangement of symbols that cannot be interpreted as an unambiguous move sequence`.

Reviewer 2: > long arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation)

I think this is not 100% clear and difficult to judge:
1. What is a *long* arrow? Are insertions written with a *short* arrow fine?
2. I think not all Delegates know what exactly an insertion is or how it looks like, so this might not be applied consistently.

Author: > Do we want the `moves written in subscript or superscript to fit between other moves`?

These are intended to cover the cases where a competitor doesn't *quite* write things in lines, but does something that is clearly intended to be read that way: https://github.com/thewca/wca-regulations/pull/575#issuecomment-407871264

After finishing the 2019 draft I want to work on an FMC [guide](https://github.com/thewca/wca-regulations/issues/236), where I want to include examples like the ones more clear.

> 1. What is a _long_ arrow? Are insertions written with a _short_ arrow fine?

Yeah, the intention is to allow simple things like in your image (where moves aren't exactly "written out of order, but definitely use small arrows). However, I wrote this before I added the "small typographic corrections" part, which I believe covers it. I've taken out "long".

> 2\. I think not all Delegates know what exactly an insertion is or how it looks like, so this might not be applied consistently.

I would like to leave this in, since it's a well-known term in most of the community (similar to "stickerless") and makes the intent pretty clear.

Reviewer 2: I still don't like this and I think this will be interpreted inconsistently.
What is a "small typographic correction"? What if I made a mistake when copying my solution and accidentally skipped 4 moves? Is this still small?
And how can the judge make sure if it's an insertion or not (especially relevant when the insertion is a sexy move or something like `R U2 R'`)?

Author: An insertion is fine if it's in-line, and I think most typographic corrections are clear in practice.

Do you have alternative suggestions for how to express that the competitor is responsible for writing a single move sequence in order, while still being lenient to the examples you gave in the other draft PR?

> - E2c2+) [EXAMPLE] Examples that are not considered a single unambiguous move sequence: multiple solution sequences without exactly one clearly indicated solution, a solution with ambiguous symbols (e.g. a letter that looks similar to "B" and "R" but is not clearly one or the other), a solution with stars or arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation), a move sequence that is not written out roughly as a series of lines in reading order, any arrangement of symbols that cannot be interpreted as an unambiguous move sequence. Exceptions to the examples above: small typographic corrections (e.g. blacking out a move and writing a move above that is clearly meant to replace it, moves written in subscript or superscript to fit between other moves).

Reviewer 3: How about the more general case of blacking out _m_ moves and writing _n_ moves above them?

Obviously, the current version allows for _m = n_ and _m > n_. What about _m < n_?

How about using plural instead to cover this (and allow for last minute adjustments): 

`blacking out moves and writing moves above that are clearly meant to replace them`

Author: ```suggestion
- E2c2+) [EXAMPLE] Examples that are not considered a single unambiguous move sequence: multiple solution sequences without exactly one clearly indicated solution, a solution with ambiguous symbols (e.g. a letter that looks similar to "B" and "R" but is not clearly one or the other), a solution with stars or arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation), a move sequence that is not written out roughly as a series of lines in reading order, any arrangement of symbols that cannot be interpreted as an unambiguous move sequence. Exceptions to the examples above: small typographic corrections (e.g. blacking out moves and writing moves above that are clearly meant to replace them, moves written in subscript or superscript to fit between other moves).
```

> - E2c2++) [ADDITION] If a competitor with little 3x3x3 Fewest Moves experience uses the incorrect type of brackets/parentheses to write a rotation, the judge may interpret the competitor's intention as the correct brackets (and the competitor should be informed), at the discretion of the Delegate. Example: {r}, <r>, or (r) may be interpreted as [r]. Note that only the bracket type may reintepreted this way, and notation such as or [R], [r]2 is still considered invalid (see [Guideline E2c4+](guidelines:guideline:E2c4+)).

Author: ```suggestion
 - E2c2++) [ADDITION] If a competitor with little 3x3x3 Fewest Moves experience uses the incorrect type of brackets/parentheses to write a rotation, the judge may interpret the competitor's intention as the correct brackets (and the competitor should be informed), at the discretion of the Delegate. Example: {r}, <r>, or (r) may be interpreted as [r]. Note that only the bracket type may reinterpreted this way, and notation such as or [R], [r u], or [r]2 is still considered invalid (see [Guideline E2c4+](guidelines:guideline:E2c4+)).
```

>  - E2c2++) [ADDITION] If a competitor with little 3x3x3 Fewest Moves experience uses the incorrect type of brackets/parentheses to write a rotation, the judge may interpret the competitor's intention as the correct brackets (and the competitor should be informed), at the discretion of the Delegate. Example: {r}, <r>, or (r) may be interpreted as [r]. Note that only the bracket type may reinterpreted this way, and notation such as or [R], [r u], or [r]2 is still considered invalid (see [Guideline E2c4+](guidelines:guideline:E2c4+)).

Reviewer: ```suggestion
  - E2c2++) [ADDITION] If a competitor with little 3x3x3 Fewest Moves experience uses the incorrect type of brackets/parentheses to write a rotation, the judge may interpret the competitor's intention as the correct brackets (and the competitor should be informed), at the discretion of the WCA Delegate. Example: {r}, <r>, or (r) may be interpreted as [r]. Note that only the bracket type may reinterpreted this way, and notation such as or [R], [r u], or [r]2 is still considered invalid (see [Guideline E2c4+](guidelines:guideline:E2c4+)).
```

Reviewer 2: Also, that's quite unclear: what is *little experience*?
I would suggest to either remove it completely (i.e. allow it for all competitors) or using the same term as in A7g (`caused by a new competitor's inexperience`).

Author: I'd really like to avoid implying that it's okay to use invalid notation.
I've changed this to "due to inexperience".

## <article-E><fewest-moves><fewestmovessolving> Article E: Fewest Moves Solving

> - E2a1+) [RECOMMENDATION] If a judge finds a submitted solution that does not appear to have information to identify the attempt, they should check both sides of the paper.

Reviewer 4: I don't think E2a1 is necessary - that kind of is just what you'd do regardless of this reg in place or not?

Author: I think we need to be explicit about what is allowed to avoid inconsistent rulings from Delegates. The incidents from this year suggest that not every Delegate is on the same page about this right now. :-(

- E2c1+) [ADDITION] If the competitor forgot to write information to identify the attempt on their intended solution paper before the time limit, they are permitted to add it in front of a judge. In this case, the competitor must ask the judge while submitting their solution and then immediately write information to identify the attempt (and make no other changes) under direct supervision of the judge. Once a competitor has finished submitting their solution (e.g. placing it in the judge's hands and allowing the judge to move on to the next competitor, placing it on the judge's table and walking away), they are not permitted to add any information. If the competitor writes anything after the time limit but before asking the judge, the attempt remains disqualified (DNF).
- E2c1++) [EXPLANATION] The competitor is not required to submit a paper with the competition name, round, and/or attempt number, but may wish to do so in order to help the organization team keep track of submitted solutions.

> - E2c1+++) [ADDITION] If a judge finds a submitted solution without information to identify the attempt (e.g. while grading the attempt), they should find out the competitor who wrote the solution in order to assign them a DNF (rather than DNS). Note that the competitor must not be allowed to write information to identify the attempt at this point, so they must receive a DNF result.

Reviewer 4: E2c1+++ - I think if they are able to find out who did the attempt, it shouldn't be DNFed so I'm not for this change

Author: I think we need to be explicit about what is allowed to avoid inconsistent rulings from Delegates. The incidents from this year suggest that not every Delegate is on the same page about this right now.

@AlbertoPdRF, would you be in favor of allowing the result to count at the discretion of the Delegate?

Reviewer 4: If you can find out who did the attempt, you know who did the attempt, you have their attempt - it shouldn't be a DNF. That's just being way too strict and I think it's unnecessary to be that strict in FMC

Author: Hmm, I just double-check and we had a unanimous WRC consensus for "Can be added in front of judge while submitting" (as opposed to the explicit alternative "Can be added until the end of the competition"): https://github.com/thewca/wca-regulations/issues/655#issuecomment-443497338

Would you be alright sticking with that, or is it possible some of us didn't take into account this case while voting?

Reviewer 4: I'd want it clarified that if the delegate can determine whose attempt it was to put the DNF in, why not put in the time? I guess when voting I completely overlooked that option. It just doesn't make any sense to me that if fit can be determined who did the attempt, that it gets entered as a DNF. If someone leaves and they can't confirm with the competitor that it is indeed their attempt, then sure, DNF the attempt, but if you can confirm it was there, I think it should be added.

Reviewer: Allowing this at the discretion of the WCA Delagate shouldn't be an option in my opinion, as we would still have the same inconsistent rulings.
